packages/proxy-addr/src/index.ts

```diff
@@ -50,38 +75,45 @@ function alladdrs(req: Req, trust: Trust): string[] {
  *
  * @param  val
  */
-function compile(val: string | string[] | number[]): (addr: string) => boolean {
+function compile(val: string | number | string[]): (addr: string, i: number) => boolean {
   let trust: string[]
   if (typeof val === 'string') trust = [val]
-  else if (Array.isArray(val)) trust = val.slice() as string[]
+  else if (typeof val === 'number') return compileHopsTrust(val)
+  else if (Array.isArray(val)) trust = val.slice()
   else throw new TypeError('unsupported trust argument')
 
   for (let i = 0; i < trust.length; i++) {
-    val = trust[i]
-    if (!Object.prototype.hasOwnProperty.call(IP_RANGES, val)) continue
+    const element = trust[i]
+    if (!isIPRangeName(element)) continue
 
     // Splice in pre-defined range
-    val = IP_RANGES[val as string]
-    trust.splice.apply(trust, [i, 1].concat(val as number[]))
-    i += val.length - 1
+    const namedRange = IP_RANGES[element]
+    trust.splice(i, 1, ...namedRange)
+    i += namedRange.length - 1
   }
   return compileTrust(compileRangeSubnets(trust))
 }
+/**
+ * Compile 'hops' number into trust function.
+ *
+ * @param hops
+ */
+function compileHopsTrust(hops: number): (_: string, i: number) => boolean {
+  return (_, i) => i < hops
+}
+
 /**
  * Compile `arr` elements into range subnets.
  */
 function compileRangeSubnets(arr: string[]) {
-  const rangeSubnets = new Array(arr.length)
-  for (let i = 0; i < arr.length; i++) rangeSubnets[i] = parseIPNotation(arr[i])
-
-  return rangeSubnets
+  return arr.map(parseIPNotation)
```

Comment:
```suggestion
  return arr.map(ip => parseIPNotation(ip))
```

To prevent parseIPNotation to accidentally chew in second argument which is an index. Right now it accepts only one, but in case its signature changes, tbud could create a bug